rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -225,4 +229,35 @@ public static boolean isValidJSONObject(String str) {
             return false;
         }
     }
+
+    public static void verifyUserAuthorization() {
+
+        if (!IS_AUTH_IN_PROGRESS) {
```

Comment:
Instead of using global variables, use the singleton pattern to create a single instance of the AsyncTask and check if it's running before re-running it https://stackoverflow.com/a/6645335/9782187.

Reply:
fixed